std/file.d

```diff
@@ -609,17 +616,79 @@ if (isConvertibleToString!R)
     static assert(__traits(compiles, append(TestAliasedString("foo"), [0, 1, 2, 3])));
 }
 
+/*********************************************
+Create file $(D name) and write $(D buffer) to it
+ensuring that the file does not exist at the time of calling.
+
+Note:
+It's not the same as consecutive calling of $(D exists) and $(D write).
+This function does not suffer from the TOCTOU (time of check time of use) problem.
+
+Params:
+    name = string or range of characters representing the file _name
+    buffer = data to be written to file
+
+Throws: $(D FileException) on error.
+ */
+void writeNewFile(R)(R name, const void[] buffer)
+if ((isInputRange!R && !isInfinite!R && isSomeChar!(ElementEncodingType!R) || isSomeString!R) &&
+    !isConvertibleToString!R)
+{
+    static if (isNarrowString!R && is(Unqual!(ElementEncodingType!R) == char))
+        writeImpl(name, name.tempCString!FSChar(), buffer, WriteType.newFile);
+    else
+        writeImpl(null, name.tempCString!FSChar(), buffer, WriteType.newFile);
+}
+
+///
+@system unittest
+{
+    import std.exception : assertThrown;
+    scope(exit)
+    {
+        assert(exists(deleteme));
+        remove(deleteme);
+    }
+
+    int[] a = [ 0, 1, 1, 2, 3, 5, 8 ];
+    writeNewFile(deleteme, a); // deleteme is the name of a temporary file
```

Comment:
I would suggest ensuring that the file does not exist at the start of the unittest, in case it was left over from a previous test run. E.g. `collectException(remove(deleteme));`.